Minesweeper.py

```diff
@@ -336,12 +351,10 @@ def uncover(self, pos):
                     if tile['covered']:
                         tile['covered'] = False
                         if tile['mine']:
-                            # DHB: again, no changing directly the state variables of another object. That should be done in member functions of the object.
-                            game.game_over = True
-                            # calls self.reveal_board on game over
+                            self.game.gamestate.game_over = True
```

Comment:
bad boy, no donut. Only gamestate may change the variables in gamestate. Use a setter instead. Querying them is fine however.